amd_openvx/openvx/hipvx/vision_kernels.cpp

```diff
@@ -54,7 +60,7 @@ Hip_CannySobel_U16_U8_3x3_L1NORM(uint dstWidth, uint dstHeight,
             goffset = (y - ly + id - 1) * srcImageStrideInBytes + (((x >> 3) - lx) << 3) + 124;
             doExtraLoad = (id < 18) ? true : false;
         }
-        if (doExtraLoad) {
+        if (doExtraLoad && goffset >= 0 && goffset < (dstHeight * srcImageStrideInBytes) - sizeof(uint2)) {
```

Comment:
Not sure what the second part of condition doing? I think this is not required because the srcImage has to be at least of size height.